### PR TITLE
Add FlowLayout tests, cleanup FlowLayout classes and fix argument validation

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/DockAndAnchorLayout.cs [hunk 1]
```diff
@@ -526,24 +504,19 @@ private static Size xGetDockedSize(IArrangedElement element, Size remainingSize,
...
-            // PreferredSize is garbage if we are not measuring.
-            return xLayout(container, /* measureOnly = */ false, out Size garbage);
+            return xLayout(container, measureOnly: false, preferredSize: out Size _);
         }
 
-        // Note: PreferredSize is only computed if measureOnly = true.
+        /// <remarks>
+        /// PreferredSize is only computed if measureOnly = true.
+        /// </remarks>
         private static bool xLayout(IArrangedElement container, bool measureOnly, out Size preferredSize)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/DockAndAnchorLayout.cs]: Let's fix the naming too, while we are at it
Propose `TryCalculatePreferredSize`

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/Layout/DockAndAnchorLayout.cs]: Fixed!

**Comment by RussKie** [comment]
Please amalgamate the last two commits and I'll merge it

**Comment by hughbe** [comment]
> Please amalgamate the last two commits and I'll merge it

Done : )

**Pull request merged**